app/state_machines/efile_submission_state_machine.rb

```diff
@@ -69,4 +67,13 @@ class EfileSubmissionStateMachine
       locale: client.intake.locale
     )
   end
+
+  after_transition(to: :resubmitted) do |submission|
+    if submission.efile_submission_transitions.where(to_state: :transmitted).count.zero?
```

Comment:
This was kind of an odd decision that I'm still not sure I'm totally on board for. 

Essentially, if we've never transmitted the submission to the IRS, there's no real reason to create a new submission and generate a new IRS id, so those I'm transitioning back to the start. That allows me to know that something that has a "previous submission" needs to include the previous submission id node when building XML for the new submission.

If this is uncomfortable, we could always create new objects even if it hasnt already been transmitted, and update the logic for determining a "previously transmitted submission"

Reply:
I updated to do this, which also helps me simplify grabbing the "previous submission" of a submission 👯